src/pages/home/ReportScreen.js

```diff
@@ -1,54 +1,109 @@
+    navigation: PropTypes.shape({
+        /* Display the drawer programmatically */
+        openDrawer: PropTypes.func.isRequired,
+    }).isRequired,
 
-const defaultProps = {
-    currentlyViewedReportID: '0',
+    /* Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        /* Route specific parameters used on this screen */
+        params: PropTypes.shape({
+            /* The ID of the report this screen should display */
+            reportID: PropTypes.string,
+        }).isRequired,
+    }).isRequired,
 };
 
-const ReportScreen = (props) => {
-    const activeReportID = parseInt(props.currentlyViewedReportID, 10);
-    if (!activeReportID) {
-        return null;
+class ReportScreen extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            error: null,
+        };
     }
 
-    return (
-        <ScreenWrapper
-            style={[
-                styles.appContent,
-                styles.flex1,
-                styles.flexColumn,
-            ]}
-        >
-            <HeaderView
-                reportID={activeReportID}
-                onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
-            />
-            <View style={[styles.dFlex, styles.flex1]}>
-                <ReportView reportID={activeReportID} />
-            </View>
-        </ScreenWrapper>
-    );
-};
+    componentDidMount() {
+        this.fetchReport();
+    }
+
+    componentDidUpdate(prevProps) {
+        // Reports changed, reset and load new data
+        if (this.reportID !== prevProps.route.params.reportID) {
+            this.fetchReport();
+        }
+    }
+
+    get canRenderMainContent() {
+        return this.reportID && !this.state.isLoading && !this.state.error;
+    }
+
+    get reportID() {
+        const params = this.props.route.params;
+        return params.reportID;
+    }
+
+    fetchReport() {
+        if (!this.reportID) { return; }
+
+        // This adds a small fade in transition when the loader appears
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+
+        this.setState({
+            isLoading: true,
+            error: null,
+        });
+
+        fetchActions(this.reportID)
+            .catch((error) => {
+                /* Todo: do something here if the user is going to be left with an empty view
+                *   E.g. Alert like "Oops, we couldn't load this conversation. Please Try again"
+                *     button Retry -> this.fetchReport() */
+                this.setState({error});
+                console.error(error);
+            })
+            .finally(() => {
+                // This adds a small fadeout transition when the loader disappears
+                LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+                this.setState({isLoading: false});
+            });
```

Comment:
I'm still fine tuning this.

My end goal is:
If there's already available data to be rendered, we don't need to wait fetching. It should still run and then if there's new data it will auto update

Looking at how currently the app works - there would always be data in Onyx due to fetchAll
So the waiting here can be removed altogether and just usa a brief loading animation to cover the FlatList rendering behind

Flat list rendering need further optimizations - that's the bottleneck on mobile native and why chat actions appear chunk by chunk - struggles with rendering (a lot better on prod build but still...). Nothing that can be done in this ticket though

Reply:
Makes sense to me. 

>just usa a brief loading animation to cover the FlatList rendering behind

Maybe something like a 300ms delay to start ?

Reply:
Ok, this was the promise usage I was talking about BTW